src/infrasys/function_data.py

```diff
@@ -0,0 +1,113 @@
+from infrasys import Component
+from typing_extensions import Annotated
+from pydantic import Field, model_validator
+from pydantic.functional_validators import AfterValidator
+from typing import NamedTuple
+import numpy as np
+
+
+class XY_COORDS(NamedTuple):
```

Comment:
Can this be renamed using PascalCase, such as `XYCoords`? Maybe that is a question for @GabrielKS on the Julia side.

Reply:
@GabrielKS, any reactions to this? Can we renamed it on the Python version?

Reply:
The rationale for `XY_COORDS` on the Julia side is that it's a `const` type alias:
```julia
const XY_COORDS = @NamedTuple{x::Float64, y::Float64}
```
and that's how we did those in the past in IS:
```julia
const CONSTANT = Float64  # still exists
const POLYNOMIAL = Tuple{Float64, Float64}  # superseded by FunctionData
const PWL = Vector{Tuple{Float64, Float64}}  # superseded by FunctionData
```

. I'm not opposed to `XYCoords` in the Python version, and there is even an argument for renaming it in the Julia version, given that in PSY we have
```julia
const MinMax = NamedTuple{(:min, :max), Tuple{Float64, Float64}}  # still exists
const UpDown = NamedTuple{(:up, :down), Tuple{Float64, Float64}}  # still exists
```
etc.